plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -57,15 +60,36 @@ public synchronized AmazonEC2 client() {
             return client;
         }
 
-        this.client = new AmazonEC2Client(buildCredentials(logger, settings), buildConfiguration(logger, settings));
+        AmazonEC2ClientBuilder builder = AmazonEC2ClientBuilder.standard()
+            .withCredentials(buildCredentials(logger, settings))
+            .withClientConfiguration(buildConfiguration(logger, settings));
+
         String endpoint = findEndpoint(logger, settings);
         if (endpoint != null) {
-            client.setEndpoint(endpoint);
+            String region = buildRegion(endpoint);
+            // TODO according to the documentation we should set the region instead of setting a endpoint
+            // See https://github.com/elastic/elasticsearch/issues/27924
+            logger.debug("Setting endpoint to [{}] with region [{}]", endpoint, region);
+            builder.setEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endpoint, region));
+        } else {
+            logger.debug("No endpoint defined. We are falling back to [ec2.us-east-1.amazonaws.com].");
```

Comment:
Normally we wouldn't use "we" in a log message like this. I suggest `Using default endpoint [...]`.